services/orion-decision/src/orion_decision/ci_matrix.py

```diff
@@ -630,9 +630,8 @@ def _parse_matrix(self, matrix, branch, is_release):
                     name = include.get("name")
 
                     if "when" in include:
-                        if include["when"].get("release") is not None:
-                            if include["when"]["release"] != is_release:
-                                continue
+                        if include["when"].get("release") and event_type != "release":
```

Comment:
This will fail for `release: false`. I think it should be: `include["when"].get("release") is (event_type == "release")`.